src/i965_avc_encoder_common.h

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright @ 2016 Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sub license, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the
+ * next paragraph) shall be included in all copies or substantial portions
+ * of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT.
+ * IN NO EVENT SHALL PRECISION INSIGHT AND/OR ITS SUPPLIERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWAR OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Authors:
+ *     Pengfei Qu <Pengfei.Qu@intel.com>
+ *
+ */
+
+#ifndef _I965_AVC_ENCODER_COMMON_H
+#define _I965_AVC_ENCODER_COMMON_H
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <stdint.h>
+#include <assert.h>
+#include "intel_driver.h"
+
+// SubMbPartMask defined in CURBE for AVC ENC
+#define INTEL_AVC_DISABLE_4X4_SUB_MB_PARTITION    0x40
+#define INTEL_AVC_DISABLE_4X8_SUB_MB_PARTITION    0x20
+#define INTEL_AVC_DISABLE_8X4_SUB_MB_PARTITION    0x10
+#define INTEL_AVC_MAX_BWD_REF_NUM    2
+#define INTEL_AVC_MAX_FWD_REF_NUM    8
+
+#define MAX_MFC_AVC_REFERENCE_SURFACES      16
+#define NUM_MFC_AVC_DMV_BUFFERS             34
+#define MAX_HCP_REFERENCE_SURFACES      8
+#define NUM_HCP_CURRENT_COLLOCATED_MV_TEMPORAL_BUFFERS             9
+
+#define INTEL_AVC_IMAGE_STATE_CMD_SIZE    128
+#define INTEL_AVC_MIN_QP    1
+#define INTEL_AVC_MAX_QP    51
+
+#define INTEL_AVC_WP_MODE_DEFAULT 0
+#define INTEL_AVC_WP_MODE_EXPLICIT 1
+#define INTEL_AVC_WP_MODE_IMPLICIT 2
+
+struct avc_param {
+
+    // original width/height
+    uint32_t frame_width_in_pixel;
+    uint32_t frame_height_in_pixel;
+    uint32_t frame_width_in_mbs;
+    uint32_t frame_height_in_mbs;
+    uint32_t frames_per_100s;
+    uint32_t vbv_buffer_size_in_bit;
+    uint32_t target_bit_rate;
+};
+
+typedef enum
+{
+    INTEL_AVC_BASE_PROFILE               = 66,
+    INTEL_AVC_MAIN_PROFILE               = 77,
+    INTEL_AVC_EXTENDED_PROFILE           = 88,
+    INTEL_AVC_HIGH_PROFILE               = 100,
+    INTEL_AVC_HIGH10_PROFILE             = 110,
+    INTEL_AVC_HIGH422_PROFILE            = 122,
```

Comment:
Please remove the  misleading definition(for example: HIGH422, HIGH444 and so on).
Otherwise maybe someone think that it will be supported in the driver.